### PR TITLE
fix: fix Checkbox not disabled when itemsLits is empty

### LINKED ISSUE #40037: Transfer 组件 checkBox 在列表为空的时候没有变成 disabled 状态
### Reproduction link

[https://ant.design/components/transfer-cn](https://ant.design/components/transfer-cn)

### Steps to reproduce

之前将 Transfer 组件由 class component 重构为 function component，在重构的过程中发现 checkBox 在列表为空的时候没有变成 disabled 状态，为了不造成其它影响，重构时还是保留了原有的逻辑

### What is expected?

列表为空的时候 checkBox 变成 disabled 状态

### What is actually happening?

列表为空的时候 checkBox 没有变成 disabled 状态

| Environment | Info |
| --- | --- |
| antd | 5.1.2 |
| React | 18.2.0 |
| System | mac |
| Browser | Chrome |

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- close #40037

### befer:

image

### after:

image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | fix: fix Checkbox not disabled when itemsLits is empty |
| 🇨🇳 Chinese | fix: 修复 checkBox 在列表为空的时候没有变成 disabled 状态 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by li-jia-nan** [comment]
顺便优化了一波代码

**Comment by li-jia-nan** [comment]
顺便优化了一波代码

File: components/config-provider/__tests__/__snapshots__/components.test.tsx.snap [hunk 1]
```diff
@@ -38402,13 +38402,14 @@ exports[`ConfigProvider components Transfer configProvider 1`] = `
       class="config-transfer-list-header"
     >
       <label
-        class="config-checkbox-wrapper config-transfer-list-checkbox"
+        class="config-checkbox-wrapper config-checkbox-wrapper-disabled config-transfer-list-checkbox"
```
Comment (zombieJ) [inline components/config-provider/__tests__/__snapshots__/components.test.tsx.snap:38405]: snapshot 居然也会变的么

File: components/transfer/index.tsx [hunk 2]
```diff
@@ -366,14 +361,15 @@ const Transfer = <RecordType extends TransferItem = TransferItem>(
       {(contextLocale) => {
         const listLocale = getLocale(contextLocale, renderEmpty || defaultRenderEmpty);
         const [leftTitle, rightTitle] = getTitles(listLocale);
+        const [leftAllLabels, rightAllLabels] = selectAllLabels;
```
Comment (zombieJ) [inline components/transfer/index.tsx:364]: 这个解构意义不大，而且申明一次变量会增加 bundle size

Comment (li-jia-nan) [reply components/transfer/index.tsx]: 已回滚

File: components/config-provider/__tests__/__snapshots__/components.test.tsx.snap (hunk 1 above)
Comment (li-jia-nan) [reply components/config-provider/__tests__/__snapshots__/components.test.tsx.snap:38405]: image

demo 确实变了

**Comment by li-jia-nan** [comment]
image

截图快照是符合期望的

**Comment by afc163** [comment]
图片

1. 先勾选一个 item。
2. 搜索，随便输入几个字符。使其搜不到东西。
3. 此时顶部会变灰，比较奇怪。

建议搜索导致的 length === 0 不去触发顶部 checkbox 的状态改变。

**Review by zombieJ — APPROVED** [review]

**Comment by zombieJ** [comment]
My Bad，到家合了没看到留言

**Comment by zombieJ** [comment]
@li-jia-nan 再发个 PR？补一下对应的测试用例~

**Pull request merged**